lib/Dialect/MSFT/ExportQuartusTcl.cpp

```diff
@@ -37,18 +41,32 @@ struct TclOutputState {
     os.indent(2);
     return os;
   };
+
+  SmallVector<Attribute> symbolRefs;
 };
 } // anonymous namespace
 
-void emitPath(TclOutputState &s, RootedInstancePathAttr path) {
-  for (auto part : path.getPath())
-    s.os << part.getValue() << '|';
+void emitPath(TclOutputState &s, RootedInstancePathAttr path,
+              SymbolCache &symCache) {
+  for (auto part : path.getPath()) {
+    auto inst = cast<msft::InstanceOp>(symCache.getDefinition(part));
+    assert(inst && "path instance must be in symbol cache");
+    auto mod = inst->getParentOfType<MSFTModuleOp>();
+    assert(mod && "path instance must have MSFT module parent");
```

Comment:
I'd rather not have this restriction. (Does it existing in the tblgen def?) Since it looks like you only need the symbol, could you use the MLIR symbol stuff instead?

Reply:
This was left from old code that looked up either a HWModuleOp or MSFTModuleOp, and had to confirm it found something. It is something we can guarantee in the tblgen def, and I can indeed just use the module symbol.

Reply:
I'm debating whether or not we want to be restrictive about where msft.instances can live. Anywhere or just in msft.modules? Should we start with restrictive then be more open in the future if necessary?

Reply:
Right now they have to support MSFTModule or HWModule parents to support partial lowering during lower-msft-to-hw (MSFTModule is lowered to HWModule first, then the MSFTInstance is lowered). Tablegen already requires the parent to be either MSFTModule or HWModule and nothing else. I think keeping that restriction for now makes sense.

Reply:
Yep, agreed.